jetkit/db/query_filter.py

```diff
@@ -0,0 +1,92 @@
+from typing import Iterable, Type
+from flask_sqlalchemy import BaseQuery
+
+
+class QueryFilter(BaseQuery):
+    """Base class for constructing query filters."""
+
+    def apply_default_filter(self) -> "QueryFilter":
+        """Modify query as desired here."""
+        raise NotImplementedError()
+
+    def get_filter(self, obj) -> bool:
+        """Permit or deny the retrieval of model objects via the instance map cache (`get()`).
+
+        This is not done via a query so it must be a code-level check.
+        """
+        raise NotImplementedError()
+
+    @property
+    def entity(self):
+        return self._primary_entity.type
+
+
+class FilteredQuery(QueryFilter):
```

Comment:
is that for SoftDeleted and similar stuff?

Reply:
Yes, the idea is to let us stack those kinds of query classes without the mixins overwriting each other